src/librustdoc/html/static/main.js

```diff
@@ -1196,31 +1196,19 @@ function hideThemeButtonState() {
             if (!next) {
                 return;
             }
-            if (hasClass(e, "impl") &&
-                (next.getElementsByClassName("method").length > 0 ||
-                 next.getElementsByClassName("associatedconstant").length > 0)) {
-                var newToggle = toggle.cloneNode(true);
-                insertAfter(newToggle, e.childNodes[e.childNodes.length - 1]);
-                // In case the option "auto-collapse implementors" is not set to false, we collapse
-                // all implementors.
-                if (hideImplementors === true && e.parentNode.id === "implementors-list") {
-                    collapseDocs(newToggle, "hide");
-                }
-            }
         };
 
         onEachLazy(document.getElementsByClassName("method"), func);
         onEachLazy(document.getElementsByClassName("associatedconstant"), func);
-        onEachLazy(document.getElementsByClassName("impl"), funcImpl);
         var impl_call = function() {};
-        // Large items are hidden by default in the HTML. If the setting overrides that, show 'em.
-        if (!hideLargeItemContents) {
-            onEachLazy(document.getElementsByTagName("details"), function (e) {
-                if (hasClass(e, "type-contents-toggle")) {
-                    e.open = true;
-                }
-            });
-        }
+        onEachLazy(document.getElementsByTagName("details"), function (e) {
+            if (!hideLargeItemContents && hasClass(e, "type-contents-toggle")) {
+                e.open = true;
+            }
+            if (!hideImplementors && hasClass(e, "implementors-toggle")) {
+                e.open = true;
+            }
```

Comment:
You have two conditions doing the same thing. What about this:

```suggestion
            if ((!hideLargeItemContents && hasClass(e, "type-contents-toggle")) ||
                 (!hideImplementors && hasClass(e, "implementors-toggle")))
            {
                 e.open = true;
            }
```

?

Reply:
I considered this and concluded it was a little harder to read since the reader has to parse a lot of parentheses. And it's not a significant byte savings. But I don't feel strongly about it - if you prefer the version with a single `if`, I'm happy to change it.

Reply:
It's indeed more difficult to read. What about:

```suggestion
            var show_large_item = !hideLargeItemContents && hasClass(e, "type-contents-toggle");
            var show_implementor = !hideImplementors && hasClass(e, "implementors-toggle");
            
            if (show_large_item || show_implementor) {
                e.open = true;
            }
```

My problem is around the `e.open = true` called twice. :)